frontend/src/services/connectionTypesService.ts

```diff
@@ -0,0 +1,63 @@
+import axios from '~/utilities/axios';
+import { ConfigMapKind } from '~/k8sTypes';
+import { ResponseStatus } from '~/types';
+
+export const fetchConnectionTypes = (): Promise<ConfigMapKind[]> => {
+  const url = `/api/connection-types`;
+  return axios
+    .get(url)
+    .then((response) => response.data)
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const fetchConnectionType = (name: string): Promise<ConfigMapKind> => {
+  const url = `/api/connection-types/${name}`;
+  return axios
+    .get(url)
+    .then((response) => response.data)
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const createConnectionType = (connectionType: ConfigMapKind): Promise<ResponseStatus> => {
+  const url = `/api/connection-types`;
+  return axios
+    .post(url, connectionType)
+    .then((response) => response.data)
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
```

Comment:
Can we do the conversion between types? Do you think it makes sense that in app we always work with the parsed object. I don't know of a time when we would need to use the raw config map.

```suggestion
export const fetchConnectionType = (name: string): Promise<ConnectionTypeConfigMapObj> => {
  const url = `/api/connection-types/${name}`;
  return axios
    .get(url)
    .then((response) => toConnectionTypeConfigMapObj(response.data))
    .catch((e) => {
      throw new Error(e.response.data.message);
    });
};

export const createConnectionType = (connectionType: ConnectionTypeConfigMap): Promise<ResponseStatus> => {
  const url = `/api/connection-types`;
  return axios
    .post(url, toConnectionTypeConfigMap(connectionType))
    .then((response) => response.data)
    .catch((e) => {
      throw new Error(e.response.data.message);
    });
};
```